src/upload.rs

```diff
@@ -0,0 +1,215 @@
+use crate::models::NewUpload;
+use crate::pinata::PinataClient;
+use flate2::read::GzDecoder;
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use forc_util::bytecode::get_bytecode_id;
+use std::fs::{self, File};
+use std::path::Path;
+use std::path::PathBuf;
+use std::process::Command;
+use tar::Archive;
+use thiserror::Error;
+use uuid::Uuid;
+
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum UploadError {
+    #[error("The project is too large to be uploaded.")]
+    TooLarge,
+    #[error("Failed to save zip file.")]
+    SaveFile,
+    #[error("Failed to copy files.")]
+    CopyFiles,
+    #[error("Invalid Forc version: {0}")]
+    InvalidForcVersion(String),
+    #[error("Failed to compile project.")]
+    FailedToCompile,
+    #[error("Failed to authenticate.")]
+    Authentication,
+    #[error("Failed to upload to IPFS.")]
+    Ipfs,
+    #[error("Failed to generate bytecode ID. Err: {0}")]
+    BytecodeId(String),
+}
+
+pub async fn handle_project_upload(
```

Comment:
doc comment for this fn would be useful

Reply:
done in 5247178ba449153e2c80057dfad266cc0c14ae44